### ISSUE TITLE
[Bug]: Checkboxes on <Form> incorrectly handled

### ISSUE THREAD

**Issue by jtsiskin** [remix-run/remix#1053]
### Which Remix packages are impacted?

- [ ] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [X] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

latest (1.0.6)

### What version of Node are you using? Minimum supported version is 14.

16

### Steps to Reproduce

```
<Form method="get">
        <label htmlFor="nike">Nike</label>
        <input type="checkbox" id="nike" name="brand" value="nike" />

        <label htmlFor="adidas">Adidas</label>
        <input type="checkbox" id="adidas" name="brand" value="adidas" />
        <button type="submit">Update</button>
</Form>
```

Check both boxes. Form will be submitted with ?brand=adidas. 

Replace `<Form>` with` <form>` to see correct behavior. It should be submitted as '?brand=nike&brand=adidas'.
See https://remix.run/docs/en/v1/guides/data-loading#url-search-params, it describes the correct behavior there. 

## Issue/Fix:
This is a similar bug as https://github.com/remix-run/remix/commit/c03046df564a06800d258816e2581ba53b835789 - using .set instead of .append. But here, it happens in useSubmitImpl - https://github.com/remix-run/remix/blob/c56761e09d1aeb27c4d3d5374759fd1bb5f79dcf/packages/remix-react/components.tsx#L1098

I've locally patched to .append and this seems to match the spec

cc @jacob-ebey

**Comment by mcansh** [comment]
fixed by #814 and should be available in the latest experimental build, stable release should be soon

**Comment by cpertile** [comment]
> fixed by #814 and should be available in the latest experimental build, stable release should be soon

Release [v1.1.0](https://github.com/remix-run/remix/releases/tag/v1.1.0) has solved this issue, can it be closed?
